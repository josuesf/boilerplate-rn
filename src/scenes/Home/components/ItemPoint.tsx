import React from 'react';
import {Image, TouchableOpacity} from 'react-native';
import {Box, Text} from '@components';
import {IProduct} from 'src/common/models/Products';
import {RFValue} from 'react-native-responsive-fontsize';
import {FormatDate} from 'src/common/utils/formats';

type Props = {
  item: IProduct;
  onPress: () => void;
};

const styleImg = {
  height: RFValue(55),
  width: RFValue(55),
  borderRadius: RFValue(10),
};

export const ItemPoint = ({item, onPress}: Props) => {
  const date = new Date(item.createdAt);
  return (
    <TouchableOpacity onPress={onPress}>
      <Box flexDirection="row" mb="s" alignItems="center" key={item.id}>
        <Box backgroundColor="gray" borderRadius={RFValue(10)} mr="s">
          <Image
            resizeMethod="auto"
            resizeMode="cover"
            style={styleImg}
            source={{uri: item.image}}
          />
        </Box>
        <Box flex={1}>
          <Text variant="h4" color="black" fontWeight="800">
            {item.product}
          </Text>
          <Text variant="h5">{FormatDate(date)}</Text>
        </Box>
        <Text variant="h4" color="black" fontWeight="800">
          <Text
            fontWeight="800"
            color={item.is_redemption ? 'red' : 'greenPrimary'}>
            {item.is_redemption ? '- ' : '+ '}
          </Text>
          {item.points}
        </Text>
        <Text ml="s" fontWeight="800">
          {'>'}
        </Text>
      </Box>
    </TouchableOpacity>
  );
};
