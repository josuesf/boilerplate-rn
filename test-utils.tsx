import React from 'react';
import {render as rntlRender} from '@testing-library/react-native';
import {ThemeProvider} from './@components';
import moment from 'moment';
import 'moment/locale/es-mx';
import {Provider} from 'react-redux';
import {store} from 'src/redux/store';

function render(ui: React.ReactElement) {
  moment.locale('es-mx');
  return rntlRender(ui, {
    wrapper: ({children}) => (
      <ThemeProvider>
        <Provider store={store}>{children}</Provider>
      </ThemeProvider>
    ),
  });
}

export * from '@testing-library/react-native';

export {render};
