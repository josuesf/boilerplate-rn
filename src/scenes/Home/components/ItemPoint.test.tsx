import * as React from 'react';
import {ItemPoint} from './ItemPoint';
import {render, screen} from 'test-utils';

it('renders ItemPoint correctly', async () => {
  const itemProduct = {
    createdAt: '2022-12-09T10:20:00.909Z',
    product: 'Fantastic Granite Bacon',
    points: 200,
    image: 'https://loremflickr.com/640/480/technics',
    is_redemption: false,
    id: '3',
  };
  const {queryByTestId} = render(
    <ItemPoint item={itemProduct} onPress={jest.fn} />,
  );
  expect(queryByTestId('item-name')).toHaveTextContent(itemProduct.product);
  expect(queryByTestId('item-date')).toHaveTextContent('9 de diciembre, 2022');
  expect(queryByTestId('item-points')).toHaveTextContent('+ 200');
  expect(screen.toJSON()).toMatchSnapshot();
});
