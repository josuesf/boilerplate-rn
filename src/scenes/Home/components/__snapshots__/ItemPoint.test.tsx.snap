// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ItemPoint correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "marginBottom": 20,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#9B9898",
            "borderRadius": 20,
            "marginRight": 20,
          },
        ]
      }
    >
      <Image
        resizeMethod="auto"
        resizeMode="cover"
        source={
          Object {
            "uri": "https://loremflickr.com/640/480/technics",
          }
        }
        style={
          Object {
            "borderRadius": 20,
            "height": 108,
            "width": 108,
          }
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "flex": 1,
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#0B0B0B",
              "fontFamily": "Avenir",
              "fontSize": 27,
              "fontWeight": "800",
              "lineHeight": 37,
            },
          ]
        }
        testID="item-name"
      >
        Fantastic Granite Bacon
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#0B0B0B",
              "fontFamily": "Avenir",
              "fontSize": 24,
              "fontWeight": "400",
              "lineHeight": 31,
            },
          ]
        }
        testID="item-date"
      >
        9 de diciembre, 2022
      </Text>
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#0B0B0B",
            "fontFamily": "Avenir",
            "fontSize": 27,
            "fontWeight": "800",
            "lineHeight": 37,
          },
        ]
      }
      testID="item-points"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#0ECD9D",
              "fontFamily": "Avenir",
              "fontSize": 31,
              "fontWeight": "800",
              "lineHeight": 39,
            },
          ]
        }
      >
        + 
      </Text>
      200
    </Text>
    <Text
      style={
        Array [
          Object {
            "color": "#020202",
            "fontFamily": "Avenir",
            "fontSize": 31,
            "fontWeight": "800",
            "lineHeight": 39,
            "marginLeft": 20,
          },
        ]
      }
    >
      &gt;
    </Text>
  </View>
</View>
`;
